incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -167,6 +158,48 @@ spec:
 		FieldShouldContainWithTimeout(configCRD, "", configSingleton, ".apiVersion", "v1alpha2", crdConversionTime)
 		FieldShouldContainMultiple(configCRD, "", configSingleton, ".spec.types", []string{"Role", "RoleBinding"})
 	})
+
+	It("should convert HNCConfig sync modes", func() {
+		// Create a tree with A as the root and B as the child
+		createSampleV1alpha1Tree()
+		// Set "propagate" for Secret in v1alpha1
+		cfg := `# temp file created by conversion_test.go
+apiVersion: hnc.x-k8s.io/v1alpha1
+kind: HNCConfiguration
+metadata:
+  name: config
+spec:
+  types:
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    mode: propagate
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    mode: propagate
+  - apiVersion: v1
+    kind: Secret
+    mode: propagate`
```

Comment:
Can we add a "Remove," "Ignore" and "foobar" (i.e. invalid) case here?

Reply:
sure working on it.

Reply:
FYI updated this test and also 2 test cases in the HNC e2e test to use CamelCase modes.